### ISSUE TITLE
Inconsistent results when adding an inconsequential `CASE` expression

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13151]
I found a discrepancy when running two semantically equivalent queries against an empty database:

---
1:
```cypher
CREATE () WITH [] AS n0 ORDER BY 0 WITH 0 AS n1 LIMIT 0 CREATE ({n3:0}) MERGE ()
-
RETURN COUNT { () } AS n
```
Produces:
```
╒═══╕
│"n"│
╞═══╡
│1  │
└───┘
```

---
2:
```cypher
CREATE () WITH [] AS n0 ORDER BY 0 WITH 0 AS n1 LIMIT 0 CREATE ({n3:(CASE WHEN true THEN 0 ELSE COUNT {()} END)}) MERGE ()
-
RETURN COUNT { () } AS n
```
Produces:
```
╒═══╕
│"n"│
╞═══╡
│0  │
└───┘
```
---

I believe the queries mentioned above are semantically equivalent and thus their result should be the same.
The only difference between them is the encapsluation of the `n3` value in a `CASE` expression which evaluates to the same value.

I encountered this issue when testing queries against the **Neo4j 5.6.0 enterprise version** in a Docker container running **alpine v.3**.
When testing the query against the community edition, the query results equal.

### Steps to reproduce
Run the aforementioned, semantically equivalent, queries against an empty database and observe that their results differ.

### Expected behavior
The queries should result in the same behavior

### Actual behavior
The first query returns `1`, whereas the second query returns `0`.

**Comment by InverseFalcon** [comment]
A LIMIT later in the query should not affect write operations earlier in the query, so we should expect 1 node to remain from the first CREATE clause.

This may be a SLOTTED runtime bug, as the second query can only execute with SLOTTED runtime, and forcing SLOTTED runtime for the first query (which would run as PIPELINED and yields 1 by default) yields the wrong result of 0.

**Comment by loveleif** [comment]
Nice catch @DominicWuest, thanks! It looks like we plan `Limit` where it should have been an `ExhaustiveLimit` in this query. I've created a ticket in our internal bug tracker for further investigation ([link](https://trello.com/c/Nnz4woSg)).

**Comment by sherfert** [comment]
Hi. We implemented a fix which will be included in 5.9 and 4.4.21.
